### ISSUE TITLE
Updating underlying projection data to generate new weights file

### ISSUE THREAD

**Issue by andersonfrailey** [open-source-economics/taxdata#51]
I just finished updating the `CBO_baseline` and `SOI_estimates` files found in the stage 1 directory to reflect the latest data from the CBO and IRS. These files are used in `Stage_I.py` to create the targets and factors used to blowup and re-weight each record in stage 2 of the extrapolation process. As I test the file I will be producing the following for review:

- The difference in baseline TaxCalc calculations using this new file and what is currently in use.
- Updated versions of `correlation.csv`, `reform_results.txt`, and `variable_stats_summary.csv` in the comparison directory of TaxCalc. 
- A comparison of these new results and CBO projections.

These will all be presented for discussion in a TaxCalc PR with the new weights file. If anyone would like a prerelease of the file I'll be happy to share it with you.

If there is any other information you would like me to include in the TaxCalc PR, please let me know.

I will also be opening a TaxData PR to update these files as well as the stage 1 and 2 extrapolation files after I finish reviewing the new weights file.

It should be noted that I will also be creating a new PUF in the near future that is based on the 2010 PUF and 2015 CPS rather than on the 2009 PUF and 2014 CPS as is currently the case. These updates are being done separately so that the source of any potential changes in output are easily found.

@MattHJensen @Amy-Xu @martinholmer @feenberg @GoFroggyRun @codykallen @jdebacker

**Comment by feenberg** [comment]
On Wed, 7 Dec 2016, andersonfrailey wrote:

> 
> It should be noted that I will also be creating a new PUF in the near future that is
> based on the 2010 PUF and 2015 CPS rather than on the 2009 PUF and 2014 CPS as is
> currently the case. These updates are being done separately so that the source of any
> potential changes in output are easily found.

Any reason not to skip to 2011 PUF and 2016 CPS directly?

dan

**Comment by andersonfrailey** [comment]
> Any reason not to skip to 2011 PUF and 2016 CPS directly?

@feenberg 
2015 was a typo I missed. I meant to put 2016 for the CPS. As for the PUF, I was not aware the 2011 file was released. I know we only have 2010, but I'll talk to Matt about possibly getting 2011.

**Comment by codykallen** [comment]
As I brought up to @andersonfrailey previously, the current distribution of various types of income and other measures does not line up well with the IRS SOI data for 2014. The attached file has these tabulations for several different measures available in the SOI and PUF datasets. Our distributions are very far off for top income groups. For example, those with AGI over $2 million, we underestimate their AGI by 10.6%, their taxable interest by 54.0%, their qualified dividends by 36.8%, their taxable IRA distributions by 50.2%, and their total itemized deductions by 53.4%. 

[TC distribution comparison.xlsx](https://github.com/open-source-economics/taxdata/files/636953/TC.distribution.comparison.xlsx)

Does the updated version address this discrepancy for high earners?

**Comment by feenberg** [comment]
On Wed, 7 Dec 2016, andersonfrailey wrote:

>
>       Any reason not to skip to 2011 PUF and 2016 CPS directly?
> 
> @feenberg
> 2015 was a typo I missed. I meant to put 2016 for the CPS. As for the PUF, I was not
> aware the 2011 file was released. I know we only have 2010, but I'll talk to Matt
> about possibly getting 2011.
>

Sorry, I think we paid for 2011 already but it hasn't come yet. 2010 is 
the last we actually have. But 2011 should be coming very soon.

dan

**Comment by andersonfrailey** [comment]
@codykallen asked:

> Does the updated version address this discrepancy for high earners?

[Here](https://github.com/andersonfrailey/Tax-Calculator/blob/UpdateWeights/Distribution%20Tests.ipynb) is a notebook examining the totals amount of each of the variables in your spreadsheet by AGI bin and what percent of the total each AGI bin has. It compares the SOI data, current tax-calc measures, and tax-calc with the new weights and blowup factors in 2014. For whatever reason it doesn't always load for me so if it's not working for you either let me know and I'll just send you the file.

Overall effects of the update seem to be mixed. All I did for this update was add the most recent SOI and CBO data. Total AGI seems to hold pretty close to SOI, but a few of the components have large discrepancies (see interest income and qualified dividends at the upper end of the spectrum).
For this update I only added the most recent CBO and SOI data. No additional targets were included. Let me try and add a couple of new targets and see if that changes anything.
